ledger-api/grpc-definitions/com/daml/ledger/api/v1/events_service.proto

```diff
@@ -0,0 +1,364 @@
+  // Required
+  string transaction_id = 1;
+
+  // The ID of the command which resulted in this transaction. Missing for everyone except the submitting party.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string command_id = 2;
+
+  // The workflow ID used in command submission. Only set if the ``workflow_id`` for the command was set.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string workflow_id = 3;
+
+  // Ledger effective time.
+  // Required
+  google.protobuf.Timestamp effective_at = 4;
+
+  // TODO: potentially missing
+  // string application_id = 6;
+  // repeated string submitters = 7; 
+}
+
+
+// Records that a contract has been created, and choices may now be exercised on it.
+//
+// Changes relative to v1.CreatedEventd
+//  - no `event_id`, as `EventOffset` structure communicates transaction scope and nesting
+//  - include `ledger_effective_time`, as that is contract metadata relevant for state evolution
+message ContractCreatedEvent {
+  reserved 1; // was 'event_id'
+
+  // The ID of the created contract.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Required
+  string contract_id = 2;
+
+  // The template of the created contract.
+  // Required
+  Identifier template_id = 3;
+
+  // The key of the created contract, if defined.
+  // Optional
+  Value contract_key = 7;
+
+  // The arguments that have been used to create the contract.
+  // Required
+  Record create_arguments = 4;
+
+  // The parties that are notified of this event. When a ``CreatedEvent``
+  // is returned as part of a transaction tree, this will include all
+  // the parties specified in the ``TransactionFilter`` that are informees
+  // of the event. If served as part of a flat transaction those will
+  // be limited to all parties specified in the ``TransactionFilter`` that
+  // are stakeholders of the contract (i.e. either signatories or observers).
+  // Required
+  repeated string witness_parties = 5;
+
+  // The signatories for this contract as specified by the template.
+  // Required
+  repeated string signatories = 8;
+
+  // The observers for this contract as specified explicitly by the template or implicitly as choice controllers.
+  // This field never contains parties that are signatories.
+  // Required
+  repeated string observers = 9;
+
+  // The agreement text of the contract.
+  // We use StringValue to properly reflect optionality on the wire for backwards compatibility.
+  // This is necessary since the empty string is an acceptable (and in fact the default) agreement
+  // text, but also the default string in protobuf.
+  // This means a newer client works with an older sandbox seamlessly.
+  // Optional
+  google.protobuf.StringValue agreement_text = 6;
+
+  // The time as of which this contract is present on the ledger.
+  google.protobuf.Timestamp ledger_effective_time = 10;
+}
```

Comment:
I'd also add another field for further contract-specific data that is not part of the Daml model (e.g., the nonce for opening the commitment in an authenticating contract ID).